### DISCUSSION TITLE
Keep bodies inside of circle

### DISCUSSION THREAD

**Discussion by KlassenKonstantin** [dyn4j/dyn4j#272] [Q&A]
Hi! I want to create bodies inside of a circle shape and keep them there. What I tried for the bounds of the circle is
```
val circle = Geometry.createPolygonalCircle(50, shape.radius)
val links = Geometry.createLinks(circle.vertices.reversed(), true)
links.forEach { link ->
    body.addFixture(link)
}
```
This works fine, but if the bodies inside of the circle move too fast, they may slip out of bounds. I tried making everything a bullet and setting the CCD to ALL, but no luck.
Is this even the right approach?

Thanks for your awesome lib! I am working on a physics powered layout for Android and gave you a shoutout a couple of times already on Twitter 🫶

**Comment by KlassenKonstantin** [comment]
Forgot to call `body.updateMass()` 🙌. All good!

**Comment by wnbittle** [comment]
Thank you for the shout outs and kind words!  I'm glad to hear you got the issue resolved.  Would you mind marking your answer as resolved?

For CCD, you should only have to flag the movable bodies as `bullet`s.  CCD works, but it's not perfect.  If you have more trouble, try converting the `Link`s into `Rectangle`s / `Polygon`s to give them more detection area and make sure the movable bodies have good detection area as well (thin shapes in general are troublesome).
